storage/resource_mount.go

```diff
@@ -0,0 +1,235 @@
+	serviceAccount := ""
+	if gm.Gcs != nil {
+		serviceAccount = gm.Gcs.ServiceAccount
+	}
+	if clusterID == "" && serviceAccount == "" {
+		return fmt.Errorf("either cluster_id or service_account must be specified to mount GCS bucket")
+	}
+	clustersAPI := compute.NewClustersAPI(ctx, m)
+	if clusterID != "" {
+		clusterInfo, err := clustersAPI.Get(clusterID)
+		if err != nil {
+			return err
+		}
+		if clusterInfo.GcpAttributes == nil {
+			return fmt.Errorf("cluster %s must have GCP attributes", clusterID)
+		}
+		if len(clusterInfo.GcpAttributes.GoogleServiceAccount) == 0 {
+			return fmt.Errorf("cluster %s must have GCP service account attached", clusterID)
+		}
+	}
+	if serviceAccount != "" {
+		cluster, err := GetOrCreateMountingClusterWithGcpServiceAccount(clustersAPI, serviceAccount)
+		if err != nil {
+			return err
+		}
+		return d.Set("cluster_id", cluster.ClusterID)
+	}
+	return nil
+}
+
+// GetOrCreateMountingClusterWithInstanceProfile ...
+func GetOrCreateMountingClusterWithGcpServiceAccount(
+	clustersAPI compute.ClustersAPI, serviceAccount string) (i compute.ClusterInfo, err error) {
+	clusterName := fmt.Sprintf("terraform-mount-gcs-%x", md5.Sum([]byte(serviceAccount)))
+	cluster := getCommonClusterObject(clustersAPI, clusterName)
+	cluster.GcpAttributes = &compute.GcpAttributes{GoogleServiceAccount: serviceAccount}
+	return clustersAPI.GetOrCreateRunningCluster(clusterName, cluster)
+}
+
+// ResourceDatabricksMount mounts using given configuration
+func ResourceDatabricksMount() *schema.Resource {
+	tpl := GenericMount{}
+	scm := common.StructToSchema(tpl, func(s map[string]*schema.Schema) map[string]*schema.Schema {
+		s["source"] = &schema.Schema{
+			Type:     schema.TypeString,
+			Computed: true,
+		}
+
+		s["uri"].ConflictsWith = []string{"abfss", "wasbs", "s3", "adl", "gcs"}
+		s["extra_configs"].ConflictsWith = []string{"abfss", "wasbs", "s3", "adl", "gcs"}
+		s["abfss"].ConflictsWith = []string{"uri", "extra_configs", "wasbs", "s3", "adl", "gcs"}
+		s["wasbs"].ConflictsWith = []string{"uri", "extra_configs", "abfss", "s3", "adl", "gcs"}
+		s["s3"].ConflictsWith = []string{"uri", "extra_configs", "wasbs", "abfss", "adl", "gcs"}
+		s["adl"].ConflictsWith = []string{"uri", "extra_configs", "wasbs", "s3", "abfss", "gcs"}
+		blocks := []string{"abfss", "wasbs", "s3", "adl", "gcs"}
+		// TODO: remove this when depricating other mount types (will require to add `force_new` annotation to structs as well)
+		for _, nm := range blocks {
+			s[nm].DiffSuppressFunc = common.MakeEmptyBlockSuppressFunc(nm)
+		}
+		// TODO: We need to have a validation function that will check that source isn't empty if other blocks aren't specified
+
+		return s
+	})
+
+	r := commonMountResource(tpl, scm)
+	r.CreateContext = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		// TODO: convert data into struct here & pass instead of converting in function itself? it would be required for GS & others
```

Comment:
yes, and create the `MountPreprocessor` interface, which will be added on S3 & GCS:

```
type MountPreprocessor interface {
   GetMountingClusterID(gm GenericMount) (string, error)
}
```